### PR TITLE
support for pd.CategoricalDtype to determine categorical features in dataframe

### PR DESCRIPTION
**Pull request by bitsbuffer**
This PR contains changes for support of pd.CategoricalDtype to determine categorical features in dataframe along with dtype="object".

### CODE REVIEW

File: tests/test_creation/test_combine_with_reference_feature.py
Comment (solegalli) [inline tests/test_creation/test_combine_with_reference_feature.py:117]: combine with feature reference does not operate over categorical variables. i would not modify this tests.

File: tests/test_creation/test_mathematical_combination.py
Comment (solegalli) [inline tests/test_creation/test_mathematical_combination.py:104]: same here, this test does not operate over categorical variables. I would try not to modify this test.

**Review by solegalli — CHANGES_REQUESTED** [review]
Hi @bitsbuffer, thank you so much for the PR. This is looking good. 

I have a few questions / requests:

- can we add a test to the specific function that you modified here: https://github.com/solegalli/feature_engine/blob/master/tests/test_variable_manipulation.py#L57
- why did you modify `df_vartypes` and `df_numeric` in the pytest config? Those dataframes are not used by the categorical encoders. I would modify either of the dataframes with `enc` in the name, and make sure that the categorical encoders tests pass.
- there are 2 other transformers that operate with categorical variables, the sklearnwrapper and the  SelectByTargetMeanPerformance. Can we make sure that those still work as expected after the changes?

Thanks a lot!

File: tests/conftest.py
Comment (bitsbuffer) [reply tests/conftest.py:34]: I checked this right now, it seems you are correct there is no need to modify dtype here. I will update this and required test case (test_variable_names_when_df_cols_are_integers)

**Comment by solegalli** [comment]
fyi #256 introduces a lot of changes to categorical encoders and tests. So we need rebase that branch and make the PR to check_estimator branch, or alternatively, wait until the other PR is merged.

Cheers

**Comment by solegalli** [comment]
Hey @bitsbuffer 

I've now merged #256 into master. Would you be able to rebase master into this branch, and continue with the changes? If you need support with finding with df need to be modified in the tests, let me know

Cheers
Sole

**Comment by bitsbuffer** [comment]
thanks @solegalli for the information. I will do the appropriate changes as suggested by you.

**Pull request closed without merging**
